src/useNavigationBuilder.tsx

```diff
@@ -49,10 +51,13 @@ const getRouteConfigsFromChildren = <ScreenOptions extends object>(
     );
   }, []);
 
-export default function useNavigationBuilder<ScreenOptions extends object>(
-  router: Router<any>,
-  options: Options
-) {
+export default function useNavigationBuilder<
+  ScreenOptions extends object,
+  Options extends DefualtOptions = DefualtOptions
+>(router: RouterHelper<any, Options>, options: Options) {
+  const { current: currentRouter } = React.useRef<Router<any>>(
```

Comment:
Since there's always going to be one router, I think no need to say `currentRouter`, `router` looks nicer 😛